lib/ansible/modules/cloud/amazon/elasticache_snapshot.py

```diff
@@ -0,0 +1,236 @@
+#!/usr/bin/python
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+ANSIBLE_METADATA = {'status': ['preview'],
+                    'supported_by': 'community',
+                    'version': '1.0'}
+
+DOCUMENTATION = """
+---
+module: elasticache
+short_description: Manage cache snapshots in Amazon Elasticache.
+description:
+  - Manage cache snapshots in Amazon Elasticache.
+  - Returns information about the specified snapshot.
+version_added: "2.3"
+author: "Sloane Hertel (@s-hertel)"
+options:
+  name:
+    description:
+      - The name of the snapshot we want to create, copy, delete, or get information regarding
+    type: string
+    required: yes
+  state:
+    description:
+      - Actions that will create, destroy, or copy a snapshot.
+    choices: ['present', 'absent', 'copy']
+  replication_id:
+    description:
+      - The name of the existing replication group to make the snapshot.
+    type: string
+    required: no
+    default: null
+  cluster_id:
+    description:
+      - The name of an existing cache cluster in the replication group to make the snapshot.
+    type: string
+    required: no
+    default: null
+  target:
+    description:
+      - The name of a snapshot copy
+    type: string
+    required: no
+    default: null
+  bucket:
+    description:
+      - The s3 bucket to which the snapshot is exported
+    type: string
+    required: no
+    default: null
+"""
+
+EXAMPLES = """
+# Note: None of these examples set aws_access_key, aws_secret_key, or region.
+# It is assumed that their matching environment variables are set.
+---
+- hosts: localhost
+  connection: local
+  tasks:
+    - name: 'Create a snapshot'
+      elasticache_snapshot:
+        name: 'test-snapshot'
+        state: 'present'
+        cluster_id: '{{ cluster }}'
+        replication_id: '{{ replication }}'
+"""
+
+RETURN = """
+elasticache:
```

Comment:
Instead of nesting under the `elasticache` key, you can pass the info back as `exit_json(changed=[whatever], **camel_dict_to_snake_dict(response))`. That would change usage from:

```
- elasticache_snapshot: [args]
  register: snapshot_info
- debug: var=snapshot_info.elasticache.[things]

# to this
- debug: var=snapshot_info.[things]
```

For facts modules, it makes sense to nest the returns, as those are returned and automatically entered to the namespace without the user `register`ing them.